### ISSUE TITLE
Typescript compile error due to rxjs hard dependency include

### ISSUE THREAD

**Issue by corvinrok** [bertrandg/angular-split#42]
When including `angular-split` in a local package.json, this module no longer compiles successfully against the latest typescript.  Typescript 2.4.1 now enforces stricter checking and the following  error is presented against he `rxjs` hard inclusion that is part of `angular-split`:

```
ERROR in C:\git\std_default_project\node_modules\angular-split\node_modules\rxjs\Subject.d.ts
(16,22): error TS2415: Class 'Subject<T>' incorrectly extends base class 'Observable<T>'.
  Types of property 'lift' are incompatible.
    Type '<R>(operator: Operator<T, R>) => Observable<T>' is not assignable to type '<R>(operator: Operator<T, R>) => Observable<R>'.
      Type 'Observable<T>' is not assignable to type 'Observable<R>'.
        Type 'T' is not assignable to type 'R'.
```

`rxjs` (5.4.2) has already released a fix for this (https://github.com/ReactiveX/rxjs/pull/2722). However, since `angular-split` brings along its own version of `rxjs` (5.0.2) (which includes the issue), building a project with this `angular-split` module now fails when run against TS 2.4.1

**Comment by mwent-cray** [comment]
+1 here

**Comment by b-connect** [comment]
Should be fixed with #44
